velox/functions/prestosql/tests/TransformKeysTest.cpp

```diff
@@ -0,0 +1,241 @@
+
+TEST_F(TransformKeysTest, basic) {
+  vector_size_t size = 1'000;
+  auto input = makeRowVector({
+      makeMapVector<int64_t, int32_t>(
+          size,
+          [](auto row) { return row % 5; },
+          [](auto row) { return row % 7; },
+          [](auto row) { return row % 11; },
+          nullEvery(13)),
+  });
+  registerLambda(
+      "plus5",
+      rowType("x", BIGINT(), "unused", INTEGER()),
+      input->type(),
+      "x + 5");
+
+  auto result =
+      evaluate<MapVector>("transform_keys(c0, function('plus5'))", input);
+
+  auto expectedResult = makeMapVector<int64_t, int32_t>(
+      size,
+      [](auto row) { return row % 5; },
+      [](auto row) { return row % 7 + 5; },
+      [](auto row) { return row % 11; },
+      nullEvery(13));
+  assertEqualVectors(expectedResult, result);
+
+  registerLambda(
+      "key+value",
+      rowType("k", BIGINT(), "v", INTEGER()),
+      input->type(),
+      "k + v");
+
+  result =
+      evaluate<MapVector>("transform_keys(c0, function('key+value'))", input);
+
+  expectedResult = makeMapVector<int64_t, int32_t>(
+      size,
+      [](auto row) { return row % 5; },
+      [](auto row) { return row % 7 + row % 11; },
+      [](auto row) { return row % 11; },
+      nullEvery(13));
+  assertEqualVectors(expectedResult, result);
+}
+
+TEST_F(TransformKeysTest, duplicateKeys) {
+  vector_size_t size = 1'000;
+  auto input = makeRowVector({
+      makeMapVector<int64_t, int32_t>(
+          size,
+          [](auto row) { return row % 5; },
+          [](auto row) { return row % 7; },
+          [](auto row) { return row % 11; },
+          nullEvery(13)),
+  });
+  registerLambda(
+      "mod2",
+      rowType("x", BIGINT(), "unused", INTEGER()),
+      input->type(),
+      "x % 2");
+
+  VELOX_ASSERT_THROW(
+      evaluate<MapVector>("transform_keys(c0, function('mod2'))", input),
+      "Duplicate map keys are not allowed");
+}
+
+TEST_F(TransformKeysTest, differentResultType) {
+  vector_size_t size = 1'000;
+  auto input = makeRowVector({
+      makeMapVector<int64_t, int32_t>(
+          size,
+          [](auto row) { return row % 5; },
+          [](auto row) { return row % 7; },
+          [](auto row) { return row % 11; },
+          nullEvery(13)),
+  });
+  registerLambda(
+      "oneTenth",
+      rowType("x", BIGINT(), "unused", INTEGER()),
+      input->type(),
+      "x::double * 0.1");
+
+  auto result =
+      evaluate<MapVector>("transform_keys(c0, function('oneTenth'))", input);
+
+  auto expectedResult = makeMapVector<double, int32_t>(
+      size,
+      [](auto row) { return row % 5; },
+      [](auto row) { return (row % 7) * 0.1; },
+      [](auto row) { return row % 11; },
+      nullEvery(13));
+  assertEqualVectors(expectedResult, result);
+}
+
+// Test different lambdas applied to different rows.
+TEST_F(TransformKeysTest, conditional) {
+  vector_size_t size = 1'000;
+
+  // Make 2 columns: the map to transform and a boolean that decided which
```

Comment:
can we also have a test with 2 lambdas with different types returned and confirm it fails.

Reply:
> can we also have a test with 2 lambdas with different types returned and confirm it fails.

I don't think this testing should apply to each lambda function. Instead, I suggest to add tests to make sure FunctionVector cannot be created to store lambda of different types. I'll do that in a separate PR.

Reply:
#2255 adds sanity checks to switch expression